src/components/counter.js

```diff
@@ -3,93 +3,90 @@
 import React, { Component } from "react";
 import { bindActionCreators } from "redux";
 import { connect } from "react-redux";
-import { StyleSheet, Text, View } from "react-native";
+import { StyleSheet, Text, View, FlatList } from "react-native";
 import { Button } from "react-native-elements";
 import {
   counterIncrement,
   counterDecrement,
-  counterIncrement2,
-  counterDecrement2
+  newCounter,
+  closeCounter
 } from "../actions/counter";
+import uuid from "uuid";
 
 const styles = StyleSheet.create({
-  counter_row_first: {
-    padding: 20,
-    flexDirection: "row",
-    justifyContent: "center"
-  },
   counter_row: {
-    padding: 20,
+    padding: 5,
     flexDirection: "row",
-    justifyContent: "center"
+    justifyContent: "center",
+    alignItems: "stretch"
   },
   counter_col: {
     flex: 1,
     flexDirection: "column"
   },
   text: {
     fontFamily: "Roboto-Light",
-    fontSize: 40,
+    fontSize: 20,
     flexDirection: "row",
     textAlign: "center",
     margin: 10
+  },
+  close_button: {
+    flexDirection: "row",
+    textAlign: "center"
   }
 });
 
-export class Learning_Project extends Component {
+class CounterRow extends Component {
   render() {
     return (
-      <View style={styles.counter_col}>
-        <View style={styles.counter_row_first}>
-          <Button
-            large
-            title="Increment +"
-            color="green"
-            onPress={this.props.actions.counterIncrement}
-          />
-          <Text style={styles.text}>{this.props.count}</Text>
-          <Button
-            large
-            title="Decrement -"
-            onPress={this.props.actions.counterDecrement}
-          />
-        </View>
-        <View style={styles.counter_row}>
-          <Button
-            large
-            title="Increment +"
-            color="green"
-            onPress={this.props.actions.counterIncrement2}
-          />
-          <Text style={styles.text}>{this.props.count2}</Text>
-          <Button
-            large
-            title="Decrement -"
-            color="red"
-            onPress={this.props.actions.counterDecrement2}
-          />
-        </View>
+      <View style={styles.counter_row}>
+        <Button
+          title="Increment +"
+          color="green"
+          onPress={() => {
+            this.props.actions.counterIncrement(this.props.index);
+          }}
+        />
+        <Text style={styles.text}>
+          {this.props.data[this.props.index].count}
+        </Text>
+        <Button
+          title="Decrement -"
+          color="red"
+          onPress={() => {
+            this.props.actions.counterDecrement(this.props.index);
+          }}
+        />
+        <Button
+          title="X"
+          color="black"
+          onPress={() => {
+            this.props.actions.closeCounter(this.props.index);
+          }}
+        />
       </View>
     );
   }
 }
 
 const mapStateToProps = state => ({
```

Comment:
Components shouldn't know about Redux. So they shouldn't have a connect function. Remember to just pass the required actions and data through props.